app/models/user.rb

```diff
@@ -54,13 +48,19 @@ class User < ApplicationRecord
 
   after_save :upload_redcap_details
 
-  def do_validate_kin_email?
-    next_of_kin_needed_to_register? || (persisted? && !is_parent)
+  def kin_details_and_child_details_on_create
+    if is_parent == false
+      validates_presence_of :kin_first_name, :kin_family_name
+      validates_format_of :kin_email, with: Devise::email_regexp
+    elsif is_parent == true
+      validates_presence_of :child_first_name, :child_family_name
+    end
   end
 
-  def kin_details_and_child_details
+  def kin_details_and_child_details_on_update
     if is_parent == false
       validates_presence_of :kin_first_name, :kin_family_name, :kin_contact_no
```

Comment:
I'm not sure why we're checking for the presence of `:kin_contact_no` only on update and not at creation, could you remind me? 🙏 
Assuming this is the correct behaviour,  this validator and the previous are the same except for that field. Maybe it'd be better to factor it out: have a validator for `kin_details_and_child_details` that ignores field `:kin_contact_no` and a separate validator for `:kin_contact_no` alone (called only on update).

Reply:
>I'm not sure why we're checking for the presence of :kin_contact_no only on update and not at creation

Because the registration page (where the `User` is created) doesn't contain the "Kin Contact No" field, but the page to update the user's details does

>Maybe it'd be better to factor it out

It crossed my mind because it's [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)er, but I think it's better to keep the two functions separate seeing as the two pages are separate too. Otherwise, in the process of changing how validation works on one page, you need to constantly ask, "will it continue to work properly for this other page too?"